### PR TITLE
Feature/testing strategy #194

### LINKED ISSUE #186: Update documentation to mention node 8.5
We have found that we use a fileCopy feature introduced in node 8.5 therefore we need to update the documentation to say that at least node 8.5 is needed.

### LINKED ISSUE #194: Document the testing strategy
To make it easier for a new developer, we should document the testing strategy. i.e. how we decide whether to do a standard unit test or a snapshot or an e2e test.

### PR DESCRIPTION
**Pull request by ray-millward-tessella**
## Description

Added a testing strategy section as well as a skeleton for the git workflow.

## Testing instructions

Add a set up instructions describing how the reviewer should test the code
- Review code
- Check Travis build
- Review changes to test coverage
- build docs and read

## Agile board tracking

connect to #194 
connect to #186 

closes #194
closes #186

### CODE REVIEW

File: docs/source/SMG/workflow.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
+Developer Workflow
+=====================================
+
+Workflow
+^^^^^^^^^^^^^^^^^^^^^^
+
+#. Select a story from the waffle board.
```
Comment (btester271828) [inline docs/source/SMG/workflow.rst]: maybe a link to the waffle board here?

File: docs/source/SMG/workflow.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
+Developer Workflow
+=====================================
+
+Workflow
```
Comment (btester271828) [inline docs/source/SMG/workflow.rst]: maybe link to maintenance section for summary of commands and useful URLs?

File: docs/source/SMG/workflow.rst (hunk 1 above)
Comment (ray-millward-tessella) [reply docs/source/SMG/workflow.rst]: Good idea, I'll update the story about adding a workflow page.

File: docs/source/SMG/workflow.rst (hunk 2 above)
Comment (ray-millward-tessella) [reply docs/source/SMG/workflow.rst]: Same here, I'll add it to the story.

**Pull request merged**